application/src/main/java/org/thingsboard/server/actors/ruleChain/DefaultTbContext.java

```diff
@@ -172,8 +174,27 @@ public void tellSelf(TbMsg msg, long delayMs) {
 
     @Override
     public void input(TbMsg msg, RuleChainId ruleChainId) {
-        msg.pushToStack(nodeCtx.getSelf().getRuleChainId(), nodeCtx.getSelf().getId());
-        nodeCtx.getChainActor().tell(new RuleChainInputMsg(ruleChainId, msg));
+        if (!msg.isValid()) {
+            return;
```

Comment:
as we are in the void method, the result of the method is supposed to use some callbacks. I believe if we just drop the processing without calling anyone, the rule-engine batch processing stratety will face the timeout all the time instead of fail-fast behaviour.